### PR TITLE
mgmt,  imagebuilder, add live tests

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#35316](https://github.com/Azure/azure-sdk-for-java/issues/35316)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/imagebuilder/azure-resourcemanager-imagebuilder/src/test/java/com/azure/resourcemanager/imagebuilder/ImageBuilderManagerTests.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                .create();
+            // @embedEnd
+            imageTemplate.refresh();
+            Assertions.assertEquals(imageTemplate.name(), templateName);
+            Assertions.assertEquals(imageTemplate.name(), imageBuilderManager.virtualMachineImageTemplates().getById(imageTemplate.id()).name());
+            Assertions.assertTrue(imageBuilderManager.virtualMachineImageTemplates().listByResourceGroup(resourceGroupName).stream().findAny().isPresent());
+        } finally {
+            if (imageTemplate != null) {
+                imageBuilderManager.virtualMachineImageTemplates().deleteById(imageTemplate.id());
+            }
```
Comment (weidongxu-microsoft) [inline sdk/imagebuilder/azure-resourcemanager-imagebuilder/src/test/java/com/azure/resourcemanager/imagebuilder/ImageBuilderManagerTests.java:138]: Delete the identity after this.

Does it require a user-assigned identity? Can system-assigned identity work?

Comment (v-hongli1) [reply sdk/imagebuilder/azure-resourcemanager-imagebuilder/src/test/java/com/azure/resourcemanager/imagebuilder/ImageBuilderManagerTests.java:139]: 1. Yes, user-assigned identity is required to use imagebuilder, system-assigned identity is not supported.
2. `Delete the identity after this.` has been fixed in the new version.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**